discovery-provider/src/tasks/playlists.py

```diff
@@ -322,4 +339,8 @@ def parse_playlist_event(
         playlist_record.upc = helpers.bytes32_to_str(event_args._playlistUPC)
 
     playlist_record.updated_at = block_datetime
+
+    if not all_required_fields_present(Playlist, playlist_record):
+        raise f"Missing required fields for Playlist in {playlist_record}"
```

Comment:
to keep the scope narrower, does it make sense for this to raise `EntityMissingRequiredFieldError` and then the caller to only pass on `EntityMissingRequiredFieldError` and re-raise all other errors?

Reply:
ahh this makes a lot of sense!

Reply:
oh, i remember, the reason i didn't raise this here is because the block hash, tx hash and other details needed to construct the error are missing in this scope but available in the main processing scope.